core/utils.py

```diff
@@ -91,6 +92,28 @@ class ExplorationConversionError(Exception):
     pass
 
 
+def open_file(
+        filename: str, mode: str, encoding: str='utf-8',
```

Comment:
Can you make `mode` more strict using `Literal` type?

Reply:
Sure !! I will try .